### PR TITLE
Attaching a logGroomerSidecar to dag processor

### PR DESCRIPTION
**Pull request by wookiist**
## Description
This PR attaches the logGroomer sidecar pod when using a standalone `dag processor`. This is to prevent scheduler logs from growing infinitely in the `logs` directory of that `dag processor` pod. 

In fact, one of the Airflow clusters my team uses had about `3.5 TiB` of scheduler logs accumulated in the emptyDir of a `dag processor` pod, which reduced the ephemeral-storage availability on that node to the point of pod eviction, resulting in a pod eviction.

```
airflow@airflow-test-dag-processor-78f9bfdb88-hmckb:/opt/airflow/logs$ ls
scheduler
airflow@airflow-test-dag-processor-78f9bfdb88-hmckb:/opt/airflow/logs$ du -sh
3.5T	     .
```

We haven't figured out why the standalone `dag processor` was accumulating scheduler logs, but we think it's a good idea to attach a logGroomer sidecar like any other scheduler pod or worker pod to prevent this from happening in the first place. 

In this PR, we've modified the helm chart to attach a logGroomer to the standalone `dag processor`.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
I think scheduler/* logs are mainly because standalone DAGFileProcessor has been extracted from scheduler (normally it runs there). That might be a good idea (@mhenc?) to possibly update the default log configuration so that in case dag file processor is run as standalone, the logs are created in a different folder. Not a big issue, but might be a good idea to improve it.

**Review by jedcunningham — CHANGES_REQUESTED** [review]
Can you add test coverage? Follow [this example](https://github.com/apache/airflow/blob/dcffbb4aff090e6c7b6dc96a4a68b188424ae174/tests/charts/test_scheduler.py#L656-L658).

**Comment by potiuk** [comment]
> Can you add test coverage? Follow [this example](https://github.com/apache/airflow/blob/dcffbb4aff090e6c7b6dc96a4a68b188424ae174/tests/charts/test_scheduler.py#L656-L658).

Right :)

**Comment by wookiist** [comment]
> Can you add test coverage? Follow [this example](https://github.com/apache/airflow/blob/dcffbb4aff090e6c7b6dc96a4a68b188424ae174/tests/charts/test_scheduler.py#L656-L658).

Thanks for the example! I've added the test code.

**Comment by wookiist** [comment]
@potiuk 
Additionally, I'm using remote logging to have it write to stdout, which is then passed to filebeat for loading into ES.
I thought that using remote logging doesn't leave any logs on pods like scheduler, but I'm wondering why the dag processor does?

**Comment by potiuk** [comment]
> @potiuk Additionally, I'm using remote logging to have it write to stdout, which is then passed to filebeat for loading into ES. I thought that using remote logging doesn't leave any logs on pods like scheduler, but I'm wondering why the dag processor does?

I believe it leaves logs there - but possibly they are much smaller and not that annoying. The feature to remove local logs automatically when you use remote logging has been **just recently** merged and is being released (withing next 2 weeks) as part of 2.6.0 https://github.com/apache/airflow/pull/29772

**Comment by potiuk** [comment]
Ah  my mistake.  I think local file deletion  was implemented before (but individually for ES task handler only),  The thing is that dag file processor logs are NOT logged remotly by default - anly "task" logs are. Scheduler/dag file processor logs are stored locally by default and you need to separtely configure them using Advanced configuration https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html#advanced-configuration if you want to send them to a remote logging system.

File: tests/charts/test_dag_processor.py [hunk 1]
```diff
@@ -549,3 +550,8 @@ def test_should_add_component_specific_annotations(self):
         )
         assert "annotations" in jmespath.search("metadata", docs[0])
         assert jmespath.search("metadata.annotations", docs[0])["test_annotation"] == "test_annotation_value"
+
+
+class TestDagProcessorLogGroomer(LogGroomerTestBase):
+    obj_name = "dag-processor"
+    folder = "dag-processor"
```
Comment (potiuk) [inline tests/charts/test_dag_processor.py]: This will likely fail static checks

Comment (wookiist) [reply tests/charts/test_dag_processor.py]: Thanks for the check!
Added newline at the end of the code.

**Comment by wookiist** [comment]
After a series of unit test failures, I realized that in LogGroomerTest, the `render_chart` function was not generating the dag processor yaml. This is because the dag processor is not enabled by default. 
When testing DagProcessor, I had to add `"dagProcessor": {"enabled": True}` value when testing the dagProcessor. Working on fixing the code for this.

**Comment by potiuk** [comment]
Also I moved the test file as part of speed optimization, so you will have to move it to (you will have conflict)

**Comment by wookiist** [comment]
oh sorry for noise. my mistake :( 
I wrongly rebased and pushed it.

**Comment by wookiist** [comment]
I accidentally rebased the commit, sorry.
I'm trying to continue the story in a new PR. Thank you for your understanding.

https://github.com/apache/airflow/pull/30726

**Comment by o-nikolas** [comment]
> I accidentally rebased the commit, sorry. I'm trying to continue the story in a new PR. Thank you for your understanding.
> 
> #30726

Closing this one then

**Pull request closed without merging**